### PR TITLE
created a version of HTTPRequest to use WWW class

### PR DESCRIPTION
**Pull request by AlmirKadric**
Remaining Tasks:
- [x] Cookie Container
- [x] Clean up pure C# version and fix macros

### CODE REVIEW

File: HTTPRequest/HTTPRequest.Unity.cs [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+		timeoutTimer.Start();
+
+		// Queue constructor for main thread execution
+		HTTPRequestManager.Queue(Constructor(url, contentType, postData, headers, cookies, cb));
+	}
+
+
+	//
+	private IEnumerator Constructor(string url, string contentType, byte[] postData, Dictionary<string, string> headers, CookieContainer cookies, Action<Exception, string> cb) {
+		Dictionary<string, string> headersCopy = new Dictionary<string, string>(headers);
```
Comment (christopherobin) [inline HTTPRequest/HTTPRequest.Unity.cs:50]: Seems like `headers` can be null in some cases (at least on the Gundam frontend) and Dictionary will complain about it being null resulting in some nasty warnings in the console.

**Comment by kefniark** [comment]
Not really a WIP anymore, but it looks fine

**Comment by ronkorving** [comment]
Pros and Cons to changing API?

**Comment by AlmirKadric** [comment]
@kefniark last piece done and tested, so if review is good we can merge

**Comment by AlmirKadric** [comment]
@ronkorving:
- HttpWebRequest is fully asynchronous and thus lighter whilst WWW checks on every frame as it uses unity coroutines
- WWW works on all unity platforms including XBox and PS4, whilst HttpWebRequest works on the main ones PC, macOS, iOS, Android
- HttpWebRequest will also work on vanilla .net

File: HTTPRequest/HTTPRequest.DotNet.cs [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Net;
+using System.IO;
+using System.Text;
+using System.Threading;
+
+
+public class HTTPRequest {
```
Comment (nullorvoid) [inline HTTPRequest/HTTPRequest.DotNet.cs:14]: Formatting is a little off here, I guess the switch to Windows (VS is default spaces, Tools -> Options -> Text Editor -> All Languages -> Tabs)

File: HTTPRequest/HTTPRequest.DotNet.cs [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+        request = null;
+
+        _request.Abort();
+        timeoutTimer.Dispose();
+        timeoutTimer = null;
+    }
+
+
+    // Create GET request and return it
+    public static HTTPRequest Get(string url, Dictionary<string, string> headers, CookieContainer cookies, Action<Exception, string> cb)
```
Comment (nullorvoid) [inline HTTPRequest/HTTPRequest.DotNet.cs]: At some point I would like to pull this out, I can see what you are doing here, but something over arching would make this much easier to test in the future.

File: Mage/CommandCenter/TransportClient/CommandHTTPClient.cs [hunk 4]
```diff
@@ -11,14 +11,12 @@ public class CommandHTTPClient : CommandTransportClient {
 	
 	//
 	private string endpoint;
-	private string username;
-	private string password;
+	Dictionary<string, string> headers;
```
Comment (nullorvoid) [inline Mage/CommandCenter/TransportClient/CommandHTTPClient.cs]: i know this is fine (functionally) but I really don't like the lack of "private" or "public" when reading code.

File: HTTPRequest/HTTPRequest.Unity.cs [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+				statusCode = int.Parse(request.responseHeaders["STATUS"].Split(' ')[1]);
+			}
+
+			cb(new HTTPRequestException(request.error, statusCode), null);
+			yield break;
+		}
+
+		// Otherwise check for cookies and return the response
+		Uri requestUri = new Uri(request.url);
+		PropertyInfo pinfoHeadersString = typeof(WWW).GetProperty("responseHeadersString", BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance);
```
Comment (nullorvoid) [inline HTTPRequest/HTTPRequest.Unity.cs:121]: For the Visual Studio users can we tag this with // HACK: so its easy to find

**Comment by nullorvoid** [comment]
LGTM.

File: HTTPRequest/HTTPRequest.DotNet.cs (hunk 2 above)
Comment (AlmirKadric) [inline HTTPRequest/HTTPRequest.DotNet.cs:14]: this has been addressed

**Pull request merged**